src/Functions/ReplaceRegexpImpl.h

```diff
@@ -32,10 +32,18 @@ template <bool replace_one = false>
 struct ReplaceRegexpImpl
 {
     /// Sequence of instructions, describing how to get resulting string.
-    /// Each element is either:
-    /// - substitution (in that case first element of pair is their number and second element is empty)
-    /// - string that need to be inserted (in that case, first element of pair is that string and second element is -1)
-    using Instructions = std::vector<std::pair<int, std::string>>;
+    struct Instruction
+    {
+        /// If non zero - perform substitution of n-th subpattern from the regexp match.
```

Comment:
Value `-1` means use `literal`, but `0` is still number of pattern, isn't it?

```suggestion
        /// If not negative - perform substitution of n-th subpattern from the regexp match.
```